Sources/Fullscreen/SettingsDetailsView/SettingsDetailsView.swift

```diff
@@ -0,0 +1,212 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import UIKit
+
+// MARK: - Model
+public protocol SettingsDetailsViewModel {
+    var icon: UIImage { get }
+    var title: String { get }
+    var primaryButtonStyle: Button.Style { get }
+    var primaryButtonTitle: String { get }
+    func text(for state: SettingsDetailsView.State) -> String
+    func textAlignment(for state: SettingsDetailsView.State) -> NSTextAlignment
+    func secondaryButtonTitle(for state: SettingsDetailsView.State) -> String?
+}
+
+// MARK: - Delegate
+public protocol SettingsDetailsViewDelegate: AnyObject {
+    func settingsDetailsView(_ detailsView: SettingsDetailsView, didChangeTo state: SettingsDetailsView.State, with model: SettingsDetailsViewModel)
+    func settingsDetailsView(_ detailsView: SettingsDetailsView, didTapPrimaryButtonWith model: SettingsDetailsViewModel)
+}
+
+// MARK: - View
+public class SettingsDetailsView: UIView {
```

Comment:
I think the name confuses me, this is details view for 1 setting, right? Should it maybe be `SettingDetailsView` without the "s"?

Reply:
Yes, when you select one of the settings items this view appears. Makes sense to have singular 👍